src/ng/directive/form.js

```diff
@@ -254,19 +255,46 @@ function FormController(element, attrs, $scope, $animate, $interpolate) {
    *
    * This method can be called to remove the 'ng-dirty' class and set the form to its pristine
    * state (ng-pristine class). This method will also propagate to all the controls contained
-   * in this form.
+   * in this form and to the parent form.
    *
    * Setting a form back to a pristine state is often useful when we want to 'reuse' a form after
    * saving or resetting it.
    */
   form.$setPristine = function() {
+    form.$$setPristineSelf();
+    forEach(controls, function(control) {
+      control.$setPristine();
+    });
+  };
+
+  // Private API: Sets the form to its pristine state.
+  // This method does not affect nested controls.
+  form.$$setPristineSelf = function() {
     $animate.setClass(element, PRISTINE_CLASS, DIRTY_CLASS + ' ' + SUBMITTED_CLASS);
     form.$dirty = false;
     form.$pristine = true;
     form.$submitted = false;
-    forEach(controls, function(control) {
-      control.$setPristine();
-    });
+    form.$$parentForm.$$updatePristine();
+  };
+
+  // Private API: update form pristine-ness
+  form.$$updatePristine = function() {
+    var isPristine = true,
+        controlsLength = controls.length,
+        i;
+
+    for (i = 0; i < controlsLength; i++) {
+      if (!controls[i].$pristine) {
+        isPristine = false;
+        break;
+      }
+    }
```

Comment:
Nitpicking: The above lines can be "summarized" as

`var isPristine = controls.every(function(control) { return control.$pristine; });`